src/pages/contact.js

```diff
@@ -1,84 +1,164 @@
 import React from "react"
 import { Link } from "gatsby"
-import { Container, Row, Col, Form, Button } from 'react-bootstrap';
+import { Container, Row, Col, Form, Button } from "react-bootstrap"
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
 
-import LaunchIcon from '@material-ui/icons/Launch';
-import twitterIcon from '../images/icons/twitter-icon-blue.svg'
+import LaunchIcon from "@material-ui/icons/Launch"
+import twitterIcon from "../images/icons/twitter-icon-blue.svg"
 import Layout from "../components/layout"
 import SEO from "../components/seo"
 import "../styles/pages/contact.scss"
 
-const Contact = () => (
-  <Layout pageClassName="contact-page">
-    <SEO title="Contact" />
-    <Container>
+const Contact = ({ data }) => {
+  const { title, content } = data.contentfulPage
+
+  const {
+    contactUs,
+    signUpForUpdates,
+    submitButton,
+    signUpButton,
+    formInputs,
+    contentBlocks,
+    meetupRsvpLink,
+    viewDataLink,
+  } = content
+
+  const formInputForName = name => formInputs.find(input => input.name === name)
+  const emailFormInput = formInputForName("email")
+  const nameFormInput = formInputForName("name")
+  const purposeFormInput = formInputForName("purpose")
+  const messageFormInput = formInputForName("message")
+  const signUpForUpdatesFormInput = formInputForName("signUpForUpdates")
+
+  const contentBlockForID = id =>
+    contentBlocks.find(contentBlock => contentBlock.blockId === id)
+  const twitterContentBlock = contentBlockForID("twitter")
+  const meetupContentBlock = contentBlockForID("meetup")
+  const dataSourceContentBlock = contentBlockForID("dataSource")
```

Comment:
Trying to remember... are these basically lists of things in Contentful? So we have to search through them so we're not relying on the order?

Reply:
They are arrays but I think the order is guaranteed; I did this because it seemed just as easy/less mistake-prone than relying on the right indices for specific content blocks. I can also use the indices just as easily though, wdyt?

Reply:
this is good! just trying to educate myself.